### DISCUSSION TITLE
Does this framework support automatic syncing of new strings across locales?

### DISCUSSION THREAD

**Discussion by Nickersoft** [ivanhofer/typesafe-i18n#644] [Q&A] [answered]
Hey @ivanhofer,

Been a user for some time now and love the project! One major pain point I'm always running into, however, is having to manual copy over new strings between all locales. Say I have 12 locales and add a new `myString` to `en/index.ts`. In order for the types to not break, I need to manually copy this string over to the 12 other locales (unless I wanna use a spread operator as mentioned in the docs to pull in base strings from `en/index.ts`). Otherwise the type system breaks, our CI fails, and I can't even get the strings to our localization provider. I'm curious if there's any way to support automatic syncing of new strings across locales. I know XCode does this if you modify your base .strings file. I looked through the docs but couldn't find an obvious answer, unless I missed something.

If not, curious what approaches other people have taken. Thanks!

**Comment by ivanhofer** [comment, marked as answer]
Hi @Nickersoft,

that functionality is currently not possible with the provided tools. I already have thought about this, but never had the time to implement it.
All needed functions are already available ([`exporter`](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/exporter), [`importer`](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/importer) and [`extendDictionary`](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/utils#extenddictionary)). Someone just needs to connect them, integrate it into the `generator` (probably [here](https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/generator/src/generator.mts#L72)) and add a [flag](https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/config/src/types.mts#L26) for it.

If you have a CI job that just does the syncing, then you could write a script that overrides the type of `Translation` to `BaseTranslation` just for that job to finish and push the strings. Or don't run type-checks at all during that job.
